service/app/controllers/otp/validator/OtpRequestValidator.java

```diff
@@ -39,22 +43,37 @@ private void commonValidation(Request otpRequest, boolean isOtpMandatory) {
 
   private void validateTypeAndKey(Request otpRequest) {
     Map<String, Object> requestMap = otpRequest.getRequest();
-
+    String userId = (String) requestMap.get(JsonKey.USER_ID);
     String type = (String) requestMap.get(JsonKey.TYPE);
     String key = (String) requestMap.get(JsonKey.KEY);
+    validateType(type);
+    if (StringUtils.isBlank(userId)) {
+      if (JsonKey.EMAIL.equalsIgnoreCase(type)) {
+        validateEmail(key);
+      } else if (JsonKey.PHONE.equalsIgnoreCase(type)) {

```

Comment:
Why validation is done only for  email/phone key not for prevUsedEmail and prevUsedPhone

Reply:
Its not required as we are overriding these value from the fetched user details and even if will not override these value ,this value will be masked which will fail email or phone validation